### PR TITLE
document upgrading to v4

### LINKED ISSUE #78: Upgrade guide for v4
Since v4 had some API changes and renaming, I was looked for an upgrade guide.
Only thing I found was the changelog. That could be a little bit clearer.
Are you interested in a Upgrading Documentation PR? Especially from 3.x.x to 4.x x?
If yes, I could send a PR in the hacktoberfest spirit.

### PR DESCRIPTION
**Pull request by kriswep**
As suggested in #78 this change documents the upgrade process from v3 to v4.

I added and linked a new file `UPGRADING.md`. Please let me know, if you would prefer that to be written in `README.md` directly or somewhere else. I think this will be better for future expansion, though.

This closes #78

### CODE REVIEW

File: UPGRADING.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+yarn add react-automata@^4.0.0 react-test-renderer@^16.3
+```
+
+### API changes
+
+The exported higher-order component known as `withStatechart` was renamed to `withStateMachine`. You should replace all occurences of `withStatechart` with `withStateMachine` in your code.
+
+`Action` and `State` components props were renamed more consistently. The `hide` and `show` prop from `Action` were unified in the new `is` prop. The `value` prop from `State` was renamed to `is` as well. You should change these props in your code accordingly.
+
+The `testStateChart` method  was renamed to `testStateMachine`, and its signature was changed to `Component[, { fixtures, extendedState }]` from `{ statechart[, fixtures][, extendedState] }, Component`. You need to change that accordingly. Please note, that you don't have to pass the statechart to `testStateMachine` directly, instead you should use the from `withStateMachine` enriched component.
```
Comment (MicheleBertoli) [inline UPGRADING.md]:
```suggestion
The `testStateChart` method  was renamed to `testStateMachine`, and its signature was changed to `Component[, { fixtures, extendedState }]` from `{ statechart[, fixtures][, extendedState] }, Component`. You need to change that accordingly. Please note that you don't have to pass the statechart to `testStateMachine` directly, the library will automatically extract it from the `withStateMachine` enriched component.
```

**Comment by MicheleBertoli** [comment]
This is awesome, thanks @kriswep.
I suggested a small change (cool feature!), please let me know what you think.

**Comment by kriswep** [comment]
Yeah, changed that 🚀

**Comment by MicheleBertoli** [comment]
🙌Thank you very much, @kriswep.

**Pull request merged**